src/blocked.c

```diff
@@ -97,6 +97,9 @@ void blockClient(client *c, int btype) {
         c->paused_list_node = listLast(server.paused_clients);
         /* Mark this client to execute its command */
         c->flags |= CLIENT_PENDING_COMMAND;
+    } else {
+        /* Master client should never be blocked unless pause */
+        serverAssert(!(c->flags & CLIENT_MASTER));
```

Comment:
i think we should exclude BLOCKED_MODULE too.
i know some modules block clients to do background (parallel) processing, so when that command is forwarded to the replica, they block the master client.
it's silly but it works for them, so let's not break it.
maybe there's also a reason to exclude BLOCKED_WAIT? i.e. the only ones we disallow are ones that are waiting for user input (list, zset, stream)

Reply:
I think the BLOCKED_MODULE use case is silly, but fair enough. I can't see a reason why we we would want to handle BLOCKED_WAIT, something is very weird if it's getting propagated to replicas.

Reply:
OK, `BLOCKED_MODULE` excluded, but I still believe we (and module developers) should find a way to make sure modules would not propagate block command in someday.

BTW I agree with @madolson , I can't see why we need to handle `BLOCKED_WAIT`.